src/Collection/Collection.router.ts

```diff
@@ -162,6 +174,69 @@ export class CollectionRouter extends Router {
     return fullCollection
   }
 
+  publishCollection = async (req: AuthRequest) => {
+    const id = server.extractFromReq(req, 'id')
+
+    // We are using the withCollectionExists middleware so we can safely assert the collection exists
+    const [dbCollection, dbItems] = await Promise.all([
+      Collection.findOne<CollectionAttributes>(id),
+      Item.findOrderedItemsByCollectionId(id),
+    ])
+    const remoteCollection = await collectionAPI.fetchCollection(
+      dbCollection!.contract_address
+    )
+
+    if (!remoteCollection) {
+      // This might be a problem with the graph lagging but we delegate the retry logic on the client
+      throw new HTTPError(
+        'The collection is not published yet',
+        { id },
+        STATUS_CODES.unauthorized
+      )
+    }
+
+    const items: ItemAttributes[] = [...dbItems]
+    let remoteItems: ItemFragment[] = []
+    let catalystItems: Wearable[] = []
+
+    const isMissingBlockchainItemIds = dbItems.some(
+      (item) => item.blockchain_item_id == null
+    )
+
+    if (isMissingBlockchainItemIds) {
+      const fetches = await Promise.all([
+        collectionAPI.fetchItemsByContractAddress(
+          dbCollection!.contract_address
+        ),
+        peerAPI.fetchWearables(remoteItems.map((item) => item.urn)),
+      ])
+      const updates = []
+
+      remoteItems = fetches[0]
+      catalystItems = fetches[1]
+
+      for (const [index, remoteItem] of remoteItems.entries()) {
+        updates.push(
+          Item.update(
+            { blockchain_item_id: remoteItem.blockchainId },
+            { id: dbItems[index].id }
+          )
+        )
+        items[index].blockchain_item_id = remoteItem.blockchainId
+      }
+
+      await Promise.all(updates)
+    }
+
+    return {
+      collection: await Bridge.consolidateCollections(
+        [dbCollection!],
+        [remoteCollection]
+      ),
+      items: await Bridge.consolidateItems(dbItems, remoteItems, catalystItems),
```

Comment:
Should we need to use `items` here?

Reply:
In this particular case it's the same result, but it's much better if we end up changing the endpoint. Updated!